roles/os_images/tasks/promote.yml

```
@@ -0,0 +1,68 @@
+---
+- name: Check if image suffix is provided
+  ansible.builtin.fail:
+    msg: os_images_name_suffix is empty please provide it.
+  when: os_images_name_suffix is defined and os_images_name_suffix | length == 0
+
+- name: Gather retire candidates info
+  openstack.cloud.image_info:
+    auth_type: "{{ os_images_auth_type }}"
+    auth: "{{ os_images_auth }}"
+    cacert: "{{ os_images_cacert | default(omit) }}"
+    interface: "{{ os_images_interface | default(omit, true) }}"
+    region_name: "{{ os_images_region | default(omit) }}"
+  register: retire_list
+  when: item.rename_image | default(os_images_promote) | bool
+
+- name: Ensure images for retirement exist
+  ansible.builtin.assert:
+    that:
+      - item.name in retire_list.openstack_images | map(attribute='name') | list
+    fail_msg: "The image {{ item.name[: -(os_images_name_suffix | length | int)] }} does not exist."
+  loop: "{{ os_images_list | list }}"
+  when: item.rename_image | default(os_images_promote) | bool
+
+- name: Gather promote candidates info
+  openstack.cloud.image_info:
+    auth_type: "{{ os_images_auth_type }}"
+    auth: "{{ os_images_auth }}"
+    cacert: "{{ os_images_cacert | default(omit) }}"
+    interface: "{{ os_images_interface | default(omit, true) }}"
+    region_name: "{{ os_images_region | default(omit) }}"
+  register: promote_list
+  when: item.rename_image | default(os_images_promote) | bool
+
+- name: Ensure images for promotion exist
+  ansible.builtin.assert:
+    that:
+      - item.name in promote_list.openstack_images | map(attribute='name') | list
```

The return value changed in openstack.cloud.image_info
```suggestion
      - item.name in promote_list.images | map(attribute='name') | list
```